File: typescript/week-07-error-handling/solutions.ts
// === Woche 7: Fehlerbehandlung ===

/**
 * Gibt a / b zurück, oder "Fehler: Division durch Null" wenn b === 0.
 */
export function safeDivide(a: number, b: number): number | string {
  // TODO: Prüfe ob b === 0 ist und gib den Fehler-String zurück
}

/**
 * Wandelt einen String in eine Zahl um. Bei Fehler: "Fehler: Keine gültige Zahl".
 */
export function parseAge(input: string): number | string {
  // TODO: Verwende Number() und isNaN() für die Umwandlung
}

/**
 * Gibt das erste Zeichen zurück, oder null bei leerem String.
 */
export function getFirstChar(s: string): string | null {
  // TODO: Prüfe ob der String leer ist
}

/**
 * Gibt die Länge des Strings zurück, oder 0 bei null/undefined.
 */
export function getLengthSafe(s: string | null | undefined): number {
  // TODO: Prüfe auf null/undefined und gib die Länge oder 0 zurück
}

/**
 * Validiert eine E-Mail: muss "@" enthalten und mindestens 3 Zeichen lang sein.
 */
export function validateEmail(email: string): boolean {
  // TODO: Prüfe Länge und ob "@" enthalten ist
}

File: swift/week-07-error-handling/solutions.swift
import Foundation

// 1. String von a/b oder "Fehler" bei Division durch 0
func safeDivide(a: Double, b: Double) -> String {
    fatalError("TODO: Gib a/b als String zurück, oder \"Fehler\" wenn b == 0")
}

// 2. Zahl oder "Fehler" wenn keine gültige Zahl
func parseAge(input: String) -> String {
    fatalError("TODO: Versuche input in Int umzuwandeln, gib die Zahl als String oder \"Fehler\" zurück")
}

// 3. Erstes Zeichen oder nil bei leerem String
func getFirstChar(s: String) -> Character? {
    fatalError("TODO: Gib das erste Zeichen von s zurück, oder nil wenn s leer ist")
}

// 4. Länge oder 0 bei nil
func getLengthSafe(s: String?) -> Int {
    fatalError("TODO: Gib die Länge von s zurück, oder 0 wenn s nil ist")
}

// 5. Muss "@" enthalten und >= 3 Zeichen
func validateEmail(email: String) -> Bool {
    fatalError("TODO: Gib true zurück wenn email.count >= 3 UND email \"@\" enthält")
}
